addons/website/static/src/components/dialog/add_page_dialog.js

```diff
@@ -242,6 +243,14 @@ export class AddPageTemplatePreview extends Component {
             if (this.props.isCustom) {
                 this.adaptCustomTemplate(wrapEl);
             }
+            // We need this to correctly compute the highlights size (the
+            // `ResizeObserver` that adapts the effects when a custom font
+            // is applied is not available), for now, we need a setTimeout.
+            setTimeout(() => {
+                for (const textEl of iframeEl.contentDocument?.querySelectorAll(".o_text_highlight") || []) {
+                    applyTextHighlight(textEl);
+                }
+            }, 200);
```

Comment:
@xO-Tx I moved the comment outside of the setTimeout to make it clearer. I am not sure why it is need for the add page dialog but not the snippets dialog, can you explain? But I'll merge like this for now :+1:

Reply:
> I am not sure why it is need for the add page dialog but not the snippets dialog, can you explain? But I'll merge like this for now 👍

@qsm-odoo Sorry for the late reply :pray: Actually, with the highlight public widget available, the adaptation mechanism (the "resize" part) works in the following way:

1. The public widget will (`onStart`) apply the effect on all highlights in the DOM for the first time.

2. After applying the effect, the code will also send a notification (see `"text_highlight_added"`) to link this highlight to an observer that will always adapt it `onResize` (as long as the effect is not removed).

This way, we don't have to worry about the font's application/text resize... as long as the highlight is linked to an observer.

Unfortunately, this is not the case in the page preview (We only apply the effect once, and we need everything that affects the final dimensions of the text to be fully loaded).

This is what I get with slower network:

[18.0.tst-page-preview-highlights-slow-mou.webm](https://github.com/user-attachments/assets/c667639d-796d-4bb0-b55e-808231d8517b)

Actually in the "Snippets Preview" dialog it's the same issue but this time, we set the code that applies the effects after the one [that awaits "all images" loading](https://github.com/odoo/odoo/blob/18.0/addons/web_editor/static/src/js/editor/add_snippet_dialog.js#L255-L270), which is enough for text to get its final size. I just skipped the `await` and got unadapted highlights again :grimacing: :

[18.0.tst-highlights-snippets-preview-mou.webm](https://github.com/user-attachments/assets/3de46520-d0bb-4d8d-b689-9752311be8cd)

So the best solution here is to transform the **highlight observer** part used in the public widget into a helper function (that probably should be converted to a simple `useHighlightEffect` hook when converting the public widget to Owl...)

This allows using the effects tracking/adaptation for the new pages/snippets preview content instead of `setTimeout()`.

I added the code for this in: https://github.com/odoo/odoo/pull/185820 (backported to 17.0 to handle custom page templates + I will adapt the code for "snippets preview" in the fw-port).

Just let me know if you think something needs to be changed or added there :+1: